BUILD.bazel

```diff
@@ -552,29 +562,28 @@ genrule(
         "//:python_node_manager_fbs",
         "//:redis-server",
         "//:redis-cli",
-        "//:ray_redis_module.so",
+        "//:libray_redis_module.so",
         "//:raylet",
         "//:raylet_monitor",
         "@plasma//:plasma_store_server",
     ],
-    outs = ["ray_pkg"],
+    outs = ["ray_pkg.out"],
     cmd = """
         set -x &&
-        mkdir -p python/ray &&
-        cp $(location python/ray/_raylet.so) python/ray &&
-        mkdir -p python/ray/core/src/ray/thirdparty/redis/src/ &&
-        cp $(location //:redis-server) python/ray/core/src/ray/thirdparty/redis/src/ &&
-        cp $(location //:redis-cli) python/ray/core/src/ray/thirdparty/redis/src/ &&
-        mkdir -p python/ray/core/src/ray/gcs/redis_module/ &&
-        cp $(locations //:ray_redis_module.so) python/ray/core/src/ray/gcs/redis_module/libray_redis_module.so &&
-        mkdir -p python/ray/core/src/ray/raylet/ &&
-        cp $(location //:raylet_monitor) python/ray/core/src/ray/raylet/ &&
-        mkdir -p python/ray/core/src/plasma &&
-        cp $(location @plasma//:plasma_store_server) python/ray/core/src/plasma/ &&
-        cp $(location //:raylet) python/ray/core/src/ray/raylet/ &&
-        for f in $(locations //:python_gcs_fbs); do cp $$f python/ray/core/generated/; done &&
-        mkdir -p python/ray/core/generated/ray/protocol/ &&
-        for f in $(locations //:python_node_manager_fbs); do cp $$f python/ray/core/generated/ray/protocol/; done &&
-        mv python $(location ray_pkg)
-        """,
+        WORK_DIR=$$(pwd) &&
+        cp -f $(location python/ray/_raylet.so) $$WORK_DIR/python/ray &&
+        mkdir -p $$WORK_DIR/python/ray/core/src/ray/thirdparty/redis/src/ &&
+        cp -f $(location //:redis-server) $$WORK_DIR/python/ray/core/src/ray/thirdparty/redis/src/ &&
+        cp -f $(location //:redis-cli) $$WORK_DIR/python/ray/core/src/ray/thirdparty/redis/src/ &&
+        mkdir -p $$WORK_DIR/python/ray/core/src/ray/gcs/redis_module/ &&
+        cp -f $(locations //:libray_redis_module.so) $$WORK_DIR/python/ray/core/src/ray/gcs/redis_module/ &&
+        cp -f $(location //:raylet_monitor) $$WORK_DIR/python/ray/core/src/ray/raylet/ &&
+        cp -f $(location @plasma//:plasma_store_server) $$WORK_DIR/python/ray/core/src/plasma/ &&
+        cp -f $(location //:raylet) $$WORK_DIR/python/ray/core/src/ray/raylet/ &&
+        for f in $(locations //:python_gcs_fbs); do cp -f $$f $$WORK_DIR/python/ray/core/generated/; done &&
+        mkdir -p $$WORK_DIR/python/ray/core/generated/ray/protocol/ &&
+        for f in $(locations //:python_node_manager_fbs); do cp -f $$f $$WORK_DIR/python/ray/core/generated/ray/protocol/; done &&
+        echo $$WORK_DIR > $@
+    """,
+    local = 1,
```

Comment:
@pcmoritz by setting `local = 1` here, this `genrule` can copy the files directly to the project dir. So we don't need to copy them in `bulid.sh`.

Reply:
Verified that this can still work even if `bazel build` is called from a non-root dir.
@alegithub111 nice fix!

Reply:
Awesome, this is a great find :) I'm glad we can get rid of the copy and meddling with the permissions!